### PR TITLE
feat: add documentation to all public components on the beagle android

### LINKED ISSUE #976: Add documentation to all public components on the beagle android
## Use case
Public components are not documented beagle android

## Proposal
Add javadoc documentation to all public components at the beagle android.

### PR DESCRIPTION
**Pull request by luisoliveirazup**
### Related Issues
Closes #976 

### Description and Example
Public components are not documented beagle android.

Expected: Add javadoc documentation to all public components at the beagle android.

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by diogooliveira-zup — APPROVED** [review]

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/Button.kt [hunk 1]
```diff
@@ -32,6 +32,15 @@ import br.com.zup.beagle.android.widget.RootView
 import br.com.zup.beagle.android.widget.WidgetView
 import br.com.zup.beagle.annotation.RegisterWidget
 
+/**
+ * Define a button natively using the server driven information received through Beagle
+ *
+ * @param text define the button text content.
+ * @param styleId reference a native style in your local styles file to be applied on this button.
+ * @param onPress attribute to define action when onPress
```
Comment (matheusribeirozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/Button.kt:40]:
```suggestion
 * @param onPress attribute to define actions when this component is pressed
```

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/TabBar.kt [hunk 2]
```diff
@@ -148,6 +158,15 @@ data class TabBar(
     }
 }
 
+/**
+ * Define the view has in the tab view
+ *
+ * @param title displays the text on the TabView component. If it is null or not declared it won't display any text.
+ * @param icon
+ *                  display an icon image on the TabView component.
+ *                  If it is left as null or not declared it won't display any icon.
```
Comment (matheusribeirozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/TabBar.kt:167]:
```suggestion
 * @param icon display an icon image on the TabView component. If it is left as null or not declared it won't display any icon.
```

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/TextInput.kt [hunk 3]
```diff
@@ -41,6 +41,25 @@ import br.com.zup.beagle.widget.core.TextInputType.PASSWORD
...
+ * Text Input component.
+ * @param placeholder The Placeholder is a text that is displayed when nothing has been entered in the editable
+ * text field.
+ * @param disabled Enables or disables the field.
+ * @param readOnly Check if the Input will be editable or read only.
+ * @param type This attribute identifies the type of text that we will receive in the editable text area.
+ * On Android and iOS, this field also assigns the type of keyboard that will be displayed to the us.
+ * @param hidden Enables the component to be visible or not.
+ * @param styleId This attribute receives a key that is registered in the Design System of each platform that
+ * customizes the component.
```
Comment (matheusribeirozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/TextInput.kt:58]:
```suggestion
 * @param styleId This attribute receives a key that is registered in the Design System of each platform that customizes the component.
```

**Review by uziasferreirazup — DISMISSED** [review]

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/Button.kt (hunk 1 above)
Comment (luisoliveirazup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/components/Button.kt]: change done!

**Review by diogooliveira-zup — APPROVED** [review]

**Review by uziasferreirazup — APPROVED** [review]

**Pull request merged**